index.src.html

```diff
@@ -563,37 +597,81 @@ <h3 id="integration-websockets">Integration with WebSockets</h3>
+
+      1.  {{Document}} objects are given an
+          <dfn export for="Document">address space</dfn> property, whose value
+          is an [=IP address space=].
+
+      1.  {{WorkerGlobalScope}} objects are given an
+          <dfn export for="WorkerGlobalScope">address space</dfn> property,
+          whose value is an [=IP address space=].
 
-  2.  Set {{Document/address space}} to "`private`" if the resource
-      used to instantiate the {{Document}} or {{Worker}} was delivered
-      from a [=private address=].
+      1.  [[Environment settings objects]] are given an
+          <dfn export for="environment settings object">address space</dfn>
+          accessor, which has the following implementations:
 
-  3.  Set {{Document/address space}} to "`public`" if the resource
-      used to instantiate the {{Document}} or {{Worker}} was delivered
-      from a [=public address=].
+          1.  For {{Window}} objects: return the [=Document/address space=] of
+              <var ignore>window</var>'s [=associated Document=].
+
+          1.  For {{WorkerGlobalScope}} objects: return
+              <var ignore>worker global scope</var>'s
+              [=WorkerGlobalScope/address space=].
+
+  1.  The [$create a new browsing context$] algorithm is amended with an extra
+      step in between the existing steps 19 and 20. Thus the initial
+      `about:blank` document inherits its creator's [=Document/address space=]:
+
+      > 20.  If |creator| is non-null, then set <var ignore>document</var>'s
+      >      [=Document/address space=] to to |creator|'s
+      >      [=Document/address space=].
+
+  1.  The [$initialize the Document object$] algorithm is amended with an extra
+      step in between the existing steps 10 and 11, before the new
+      {{Document}}'s [=Document/CSP list=] is initialized:
+
+      > 12.  Set <var ignore>document</var>'s [=Document/address space=] to
+      >      <var ignore>response</var>'s [=response/address space=].
+
+  1.  The [$run a worker$] algorithm is amended with two extra steps in between
+      the existing steps 14.4 and 14.5, after the new {{WorkerGlobalScope}}'s
+      referrer policy is set:
+
+      > 14.5. If |response|'s [=response/url=]'s [=url/scheme=] is a
+      >       [=local scheme=], then set |worker global scope|'s
+      >       [=WorkerGlobalScope/address space=] to <var ignore>owner</var>'s
+      >       [=WorkerGlobalScope/address space=].
+      > 14.5. Otherwise, set |worker global scope|'s
+      >       [=WorkerGlobalScope/address space=] to |response|'s
+      >       [=response/address space=].
+
+  ISSUE(WICG/cors-rfc1918#27): Find a place to insert address space inheritance
+  from the navigation initiator for `data:` URLs.
+
+  ISSUE(WICG/cors-rfc1918#27): Capture the address space of the URL creator
+  when `createObjectUrl()` is called on a blob, and inherit that when loading
+  a document from the resulting URL.
```

Comment:
Perhaps linking to the more general blob store integration bug here would be reasonable? (https://github.com/w3c/FileAPI/issues/142)

Reply:
Hmm, good point. I'll redirect this issue to the blob-specific github issue in this repo, and mention on there that we are blocked by w3c/FileAPI#142 upstream.